File: ResourceManager/Services/NoteDataService.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Design;
using ResourceManager.Models;
using ResourceManager.Services.Abstractions;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ResourceManager.Services
{
    public class NoteDataService : GenericDataService<Note>, INoteDataService
    {
        public NoteDataService(IDesignTimeDbContextFactory<ResourceManagerDBContext> contextFactory) : base(contextFactory)
        {
        }
        public override async Task<IList<Note>> GetAll()
        {
            using (ResourceManagerDBContext db = dbFactory.CreateDbContext(null!))
            {
                IList<Note> entities = await db.Notes.Include(x => x.NoteTags).ThenInclude(x => x.Tag).OrderByDescending(x => x.Date).ToListAsync();
                return entities;
            }
        }
        public async Task<int> UpdateTags(Note note, IList<Tag> tags)
        {
            using (ResourceManagerDBContext db = dbFactory.CreateDbContext(null!))
            {
                var p = db.Notes.Include(x => x.NoteTags).Single(p => p.ID == note.ID);
                db.UpdateManyToMany(p.NoteTags, tags.Select(x => new NoteTag() { NoteID = note.ID, TagID = x.ID }), x => x.TagID);

                return await db.SaveChangesAsync();
            }
        }
    }
}

File: ResourceManager/Services/NoteDataService.vb
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Design
Imports ResourceManager.Models
Imports ResourceManager.Services.Abstractions

Namespace ResourceManager.Services
    Public Class NoteDataService
        Inherits GenericDataService(Of Note)
        Implements INoteDataService
        Public Sub New(ByVal contextFactory As IDesignTimeDbContextFactory(Of ResourceManagerDBContext))
            MyBase.New(contextFactory)
        End Sub
        Public Overrides Async Function GetAll() As Task(Of IList(Of Note)) Implements IDataService(Of Note).GetAll
            Dim db = MyBase.dbFactory.CreateDbContext(Nothing)
            Using db
                Dim entities = Await db.Notes.Include(Function(x) x.NoteTags).ThenInclude(Function(x) x.Tag).OrderByDescending(Function(x) x.Date).ToListAsync()
                Return entities
            End Using
        End Function
        Public Async Function UpdateTags(ByVal note As Note, ByVal tags As IList(Of Tag)) As Task(Of Integer) Implements INoteDataService.UpdateTags
            Dim db = MyBase.dbFactory.CreateDbContext(Nothing)
            Using db
                Dim p = db.Notes.Include(Function(x) x.NoteTags).Single(Function(x) x.ID = note.ID)
                db.UpdateManyToMany(p.NoteTags, tags.Select(Function(x) New NoteTag() With {.NoteID = note.ID, .TagID = x.ID}), Function(x) x.TagID)
                Return Await db.SaveChangesAsync()
            End Using
        End Function
    End Class
End Namespace
